src/mappings/handlers/loansHandlers.ts

```diff
@@ -84,13 +83,9 @@ async function _handleLoanBorrowed(event: SubstrateEvent<LoanBorrowedEvent>): Pr
   const pool = await PoolService.getById(poolId.toString())
   if (pool === undefined) throw new Error('Pool not found!')
 
-  const currency = await CurrencyService.getOrInit(pool.currencyId.toString())
   const amount = borrowAmount.isInternal
     ? borrowAmount.asInternal.toString()
-    : borrowAmount.asExternal.quantity
-        .div(WAD)
-        .mul(borrowAmount.asExternal.settlementPrice.divn(currency.decimals))
-        .toString()
+    : borrowAmount.asExternal.quantity.div(WAD).mul(borrowAmount.asExternal.settlementPrice).toString()
```

Comment:
This won't work, I missed that in the previous PR.

If quantity is `0.5`, then `0.5*10**18 / 1*10**18 = 0`, because it's an integer and not a floating point number.

It should be fine if you swap it:

```suggestion
    : borrowAmount.asExternal.quantity.mul(borrowAmount.asExternal.settlementPrice).div(WAD).toString()
```